api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -56,15 +56,37 @@ type OpenstackDataPlaneServiceCert struct {
 	EDPMRoleServiceName string `json:"edpmRoleServiceName,omitempty"`
 }
 
+// OpenStackDataPlaneConfigMap defines the properties of the ConfigMaps used
+// by the service.
+type OpenStackDataPlaneConfigMap struct {
+	// Name of the ConfigMap
+	Name string `json:"name"`
+
+	// Required is whether the ConfigMap is required to exist
+	// +kubebuilder:validation:Optional
+	Required bool `json:"required,omitempty"`
+}
+
+// OpenStackDataPlaneSecret defines the properties of the Secrets used by
+// the service.
+type OpenStackDataPlaneSecret struct {
+	// Name of the Secret
+	Name string `json:"name"`
+
+	// Required is whether the Secret is required to exist
+	// +kubebuilder:validation:Optional
+	Required bool `json:"required,omitempty"`
+}
+
 // OpenStackDataPlaneServiceSpec defines the desired state of OpenStackDataPlaneService
 type OpenStackDataPlaneServiceSpec struct {
 	// ConfigMaps list of ConfigMap names to mount as ExtraMounts for the OpenStackAnsibleEE
 	// +kubebuilder:validation:Optional
-	ConfigMaps []string `json:"configMaps,omitempty" yaml:"configMaps,omitempty"`
+	ConfigMaps []OpenStackDataPlaneConfigMap `json:"configMaps,omitempty" yaml:"configMaps,omitempty"`
 
 	// Secrets list of Secret names to mount as ExtraMounts for the OpenStackAnsibleEE
 	// +kubebuilder:validation:Optional
-	Secrets []string `json:"secrets,omitempty"`
+	Secrets []OpenStackDataPlaneSecret `json:"secrets,omitempty"`
```

Comment:
instead of having 2 lists, maybe we can reuse the struct from AnsibleVarsFrom: https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/common.go#L28-L38

Reply:
we shoudl not do that unless we entirely rename that class

the use case we are trying to enable is the creation of optional configmaps/secrets that will contain file that will be copied to the edpm nodes 

so without removing the AnsibleVars part for he class name this would be very confusing to anyone looking at the crd.

we are not trying to pass info to ansibele via the inventory we are trying to pass file to the deployed service.

Reply:
yes, definitely we would need to rename the class